plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -625,15 +627,20 @@ protected ExecutionResult cleanupNetworkElementCommand(final IpAssocCommand cmd)
 
                 // there is only one ip in this public vlan and removing it, so
                 // remove the nic
-                if (ipsCount == 1 && !ip.isAdd()) {
-                    removeVif = true;
+                if (lastIp != null && lastIp.equalsIgnoreCase("true") && !ip.isAdd()) {
```

Comment:
True, but a lot of the other code is already using StringUtils or is moving to it. Reviewers in other PRs tend to recommend it.